tests/stdlib/tonce.nim

```diff
@@ -0,0 +1,34 @@
+discard """
+  cmd: "nim c -r --threads:on $options $file"
+  matrix: "-d:caseA; -d:caseB"
+"""
+import std/once
+
+when defined(caseA):
+  block:
+    var thr: array[0..4, Thread[void]]
```

Comment:
more idiomatic:
* `var thr: array[5, Thread[void]]`
* and as https://github.com/nim-lang/Nim/pull/16192/files#r540547170 upstream code in threads.nim should be updated